src/DeviceActivateController.js

```diff
@@ -0,0 +1,74 @@
+/*!
+ * Copyright (c) 2017, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+/* eslint max-len: [2, 160] */
+import { loc } from 'okta';
+import FormController from 'util/FormController';
+import FormType from 'util/FormType';
+import TextBox from './views/shared/TextBox';
+
+export default FormController.extend({
+  className: 'device-code-activate',
+  Model: {
+    props: {
+      userCode: ['string', true],
+      stateToken: 'string',
+    },
+    save: function() {
+      const self = this;
+      return this.doTransaction(function(transaction) {
+        return transaction.deviceActivate({
+          userCode: self.get('userCode')
+        });
+      });
+    },
+  },
+  Form: {
+    noCancelButton: true,
+    autoSave: true,
+    save: function() {
+      return loc('oform.next', 'login');
+    },
+    title: function() {
+      return loc('oie.device.code.activate.title', 'login');
```

Comment:
I removed `oie` prefix. There are few keys with `idx` prefix which are coming from backend so hard to change now. Should I just add new keys for legacy? they will be duplicates though.

Reply:
Use same key as long as possible.

> with idx prefix which are coming from backend

Are those keys used at both v1 and v2 flows?

Reply:
ideally we would like to use the same keys for both v1 and v2 flow.

@iancorcoran-okta  would it be okay to remove `idx` prefix from keys [here](https://github.com/okta/okta-core/blob/6fc7e9ac4351eae63c25aa3f3cf414d4c3a7b383/components/platformx/idx/api/src/main/java/com/okta/api/mediation/services/auth/TerminalStateTokenHelper.java#L37), so that we can use the same keys on OIE and legacy sign in widget?  cc: @larsjohansen-okta

Reply:
> Use same key as long as possible.
> 
> > with idx prefix which are coming from backend
> 
> Are those keys used at both v1 and v2 flows?

@haishengwu-okta  @larsjohansen-okta  After looking up more v1 api response does not return any keys like v2 does. So would it be okay define v1 keys on siw side? they will same value as idx keys but would be better to separate them out so there is no confusion. We can follow up later with IDX team to check if we can use the same keys and remove `idx` prefix in future.

Reply:
That `TerminalStateTokenHelper` is only used by OIE